src/components/nav/nav-controller.ts

```diff
@@ -1442,54 +1442,69 @@ export class NavController extends Ion {
       // the ElementRef of the actual ion-page created
       let pageElementRef = component.location;
 
-      // a new ComponentRef has been created
-      // set the ComponentRef's instance to its ViewController
-      view.setInstance(component.instance);
+      let promise = null;
+      if ( component.instance['_ionicProjectContent'] ) {
+        component.instance['_ionicProjectContent']().then( () => {
+          this._loadPageCont(view, navbarContainerRef, opts, component, pageElementRef, done);
+        });
+      }
+      else {
+        this._loadPageCont(view, navbarContainerRef, opts, component, pageElementRef, done);
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  private _loadPageCont(view: ViewController, navbarContainerRef: ViewContainerRef, opts: NavOptions, component: any, pageElementRef: ElementRef, done: Function) {
```

Comment:
To avoid a Promise.resolve call for the `else` case.  The view is already loaded, so we don't need to load the component.  We can easily keep this one method but this way is more performant.  I am content either way.